File: api/Models/user.model.js
const mongoose = require("mongoose");
const unipueValidator = require("mongoose-unique-validator");

const userSchema = new mongoose.Schema({
  username: {
    type: String,
    required: true,
    unique: true,
  },
  email: {
    type: String,
    required: true,
    unique: true,
  },
  password: {
    type: String,
    required: true,
  },
  date: {
    type: Date,
    default: Date.now(),
  },
  bio: {
    type: String,
    required: false,
    default: "No Bio",
  },
  mobile: {
    type: String,
    required: false,
    default: "No Number",
  },
  myposts: [
    { type: mongoose.Schema.Types.ObjectId, required: false, ref: "forums" },
  ],
  mydiaries: [
    { type: mongoose.Schema.Types.ObjectId, required: false, ref: "diaries" },
  ],
});

userSchema.set("toJSON", {
  transform: (document, returnedObject) => {
    returnedObject.id = returnedObject._id.toString();
    delete returnedObject._id;
    delete returnedObject.__v;
    delete returnedObject.password;
  },
});

//userSchema.plugin(unipueValidator, { message: "Email already in use" });

const User = mongoose.model("user", userSchema);
module.exports = User;

File: client/lib/models/user_model.dart
class UserModel {
  UserModel({
    required this.username,
    required this.email,
    required this.date,
    required this.bio,
    required this.myposts,
    required this.mydiaries,
    required this.mobile,
    required this.id,
  });
  late final String? username;
  late final String? email;
  late final String? date;
  late final String? bio;
  late final List<dynamic>? myposts;
  late final List<dynamic>? mydiaries;
  late final String? mobile;
  late final String? id;

  UserModel.fromJson(Map<String, dynamic> json) {
    username = json['username'];
    email = json['email'];
    date = json['date'];
    bio = json['bio'];
    // myposts = List.castFrom<dynamic, dynamic>(json['myposts']);
    // mydiaries = List.castFrom<dynamic, dynamic>(json['mydiaries']);
    mobile = json['mobile'];
    id = json['id'];
  }

  Map<String, dynamic> toJson() {
    final _data = <String, dynamic>{};
    _data['username'] = username;
    _data['email'] = email;
    _data['date'] = date;
    _data['bio'] = bio;
    _data['myposts'] = myposts;
    _data['mydiaries'] = mydiaries;
    _data['mobile'] = mobile;
    _data['id'] = id;
    return _data;
  }
}
